### DISCUSSION TITLE
How to add Global Properties for cpp files in VSCode?

### DISCUSSION THREAD

**Discussion by vedanta28** [microsoft/vscode-cpptools#10990] [Q&A]
Now, for using VSCode in my Mac M1, I have to basically have to keep a .vscode folder in every directory where I run my code on. Which contains the `c_cpp_properties.json`, `tasks.json`, and `settings.json files`. Is there any way to make sure that every C++ file in my Laptop follows a specific configuration if there are no local configuration files overwriting them? Even a solution which inculcates this behaviour in subdirectories is also fine.

**Comment by michelleangela** [comment]
In the user scope `settings.json` file, you can set `C_Cpp.default.*` for the C++ extension (for more info see https://code.visualstudio.com/docs/cpp/customize-default-settings-cpp#_visual-studio-code-settings). Any `C_Cpp.default.*` set in the user scope `settings.json` file should be applied to any C++ folder that does not have a local `settings.json` or `c_cpp_properties.json` filess
